benches/cuda.rs

```diff
@@ -10,17 +11,27 @@ fn random_bytes<const BYTES: usize>() -> Vec<u8> {
 
 pub fn criterion_benchmark(c: &mut Criterion) {
     let mut group = c.benchmark_group("cuda");
-    group.sample_size(500);
+    group.sample_size(100);
     group.measurement_time(Duration::from_secs(30));
 
-    let big_piece = random_bytes::<4194304>(); // 1024 * 4096
-    let expanded_ivs = random_bytes::<32768>(); // 1024 * 32
+    const NUM_PIECES: usize = 1024 * 1024;
+    let size = NUM_PIECES * 4096 / 1073741824;
+    let big_piece = random_bytes::<{4096 * NUM_PIECES}>();
```

Comment:
What was the reason for changing this to allocate 4G of memory? Seems like a lot to me.

Reply:
The idea is to allocate sufficient space that the GPU can be kept busy between the compute and data transfer. The 4G space is subdivided into sub-buffers (see https://github.com/simonatsn/sloth256-189/blob/master/src/opencl/opencl.cpp#L32), all of which operate in parallel. This allows compute and transfer to overlap. The use of 4G is somewhat arbitrary, ie a smaller amount is also probably sufficient to get most of the benefit. However, too small and the overhead will dominate. Our assumption is vast amounts of data will need to be encoded so larger is better.